client/components/SendZoopDialog.tsx

```diff
@@ -0,0 +1,155 @@
+
+    const currentUser = useGetMeQuery().data;
+
+    const [postZoop, { isLoading: isPostZoopLoading, isError, data: zoopData }] = usePostZoopMutation();
+    const {data: usersData, isLoading: isGetAllUsersLoading, error } = useGetAllUsersQuery() 
+
+    const navigate = useNavigate();
+
+    const handleSendZoopClick = () => {
+        if (!usersData) {
+            throw new Error('No users to send Zoop to');
+            // TODO: How best to render a message about this error in UI?
+        } else {
+            const [recipient] = usersData.users.filter(user => user.username === username);
+            if (recipient) {
+                postZoop({
+                    content: content,
+                    authorId: currentUser?.user.id,
+                    receiverId: recipient.id
+                })
+                .unwrap()
+                .then((payload) => {
+                    console.log('fulfilled', payload);
+                    setLastZoopSentId(payload.zoop.id)
+                })
+                .catch((error) => console.error('rejected', error));
+              
+                setZoopSent(true)
+            } else {
+                throw new Error('No such user exists. Please select a different username');
+                // TODO: How best to render this error in UI?
+            }
+            
+        }
+        
+    }
+
+    return ( //TODO: MUI suggests adding a wrapper element so that Tooltip works when Fab is disabled
+        <>
+            <Tooltip 
+                title="Send a Zoop"
+            >
+                <Fab
+                    disabled={!token}
+                    sx={{
+                        position: "absolute",
+                        bottom: 25,
+                        right: 25
+                    }}
+                    onClick={() => setOpen(true)}
+                > <SendIcon />
+                </Fab>
+            </Tooltip>
+
+            {/* Render a success dialog if Zoop was sent and successfully integrated into DB,
+                or an input dialog if Zoop has not yet been sent */}
+            {zoopSent ? (
+                <Dialog
+                    open={open}
+                    onClose={() => setOpen(false)}
+                >
+                    <DialogTitle>Zoop Sent!</DialogTitle>
+                    <DialogContent>
+                        <DialogContentText> 
+                            <Link
+                                component="button"
+                                onClick={() => navigate(`/zoops/${lastZoopSentId}`)}
+                            > 
+                                Go to this page to view your Zoop.
+                            </Link>
+                        </DialogContentText>
+
+                    </DialogContent>
+                </Dialog>
+            ) : (
+                <Dialog
+                    open={open}
+                    onClose={() => setOpen(false)}
+                >
```

Comment:
FYI this really feels like a separate component, so you might consider separating it out to make things more readable